prql-compiler/src/parser.rs

```diff
@@ -28,7 +30,21 @@ pub(crate) type PestRule = Rule;
 pub fn parse(string: &str) -> Result<Query> {
     let ast = ast_of_string(string, Rule::query)?;
 
-    ast.item.into_query().map_err(|_| unreachable!())
+    let query = ast.item.into_query().unwrap();
+
+    if let Some(ref version) = query.version {
+        check_query_version(version, &PRQL_VERSION)?;
+    }
+
+    Ok(query)
+}
+
+fn check_query_version(query_version: &VersionReq, prql_version: &Version) -> Result<()> {
+    if !query_version.matches(prql_version) {
+        bail!("This query uses version of PRQL that is not supported by your prql-compiler. You may want to upgrade the compiler.");
```

Comment:
Minor typo. I think it should read "This query uses **a** version of PRQL ...".

Reply:
@snth good spot (feel free to use the suggestion feature for something like this; i.e. 

```suggestion
        bail!("This query uses a version of PRQL that is not supported by your prql-compiler. You may want to upgrade the compiler.");
```